### ISSUE TITLE
Support - Citation Network Issues

### ISSUE THREAD

**Issue by maahutch** [iuni-cadre/Fellow1-data-citation#5]
This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.

Thank you so much for doing this! I’ll let you know when it’s done downloading (tonight if everything goes well). It’s giving me an estimate of 5-6 hours to complete download right now.

Best regards,
Nicoletta

Nicoletta Fala, Ph.D.
http://nicolettafala.com/  

From: Hutchinson, Matthew Alexander
Sent: Thursday, December 12, 2019 1:24 PM
To: Nicoletta Fala
Cc: Pentchev, Valentin; Yan, Xiaoran
Subject: RE: RE: Citation Network issues...

Hi Nicola, 

Sorry this took longer than expected, I hadn’t anticipated the trouble I’d have with a  file this size. Here is the link for the download: 

https://cs2bcdf9550be1bx4d49x996.blob.core.windows.net/mag-neo4j-nf/mag_cite_full.tar.gz 

I ended up using Microsoft azure for the storage in case you’re wondering about the url. The database was created using Neo4j 3.4 so if you’re using the latest version (3.5) you’ll need to uncomment the ‘dbms.allow_upgrade=true’ option in the config file. 

Right now the file is available for anyone to download. Once you’re done, please let me know and I can close the container. 

Let know if anything’s not working correctly, 

Thanks, 

Matthew Hutchinson | INDIANA UNIVERSITY
Data Manager
IU Network Science Institute (IUNI)
1001 E SR 45/46 Bypass | Bloomington, IN  47408-1415
Email:  maahutch@iu.edu |Phone: (812) 855-1404
Fax: (812) 856-1192

From: Nicoletta Fala <nfala@purdue.edu> 
Sent: Thursday, December 12, 2019 10:41 AM
To: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Cc: Pentchev, Valentin <vpentche@iu.edu>; Yan, Xiaoran <yan30@iu.edu>
Subject: [External] Re: RE: Citation Network issues...

This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.

I do not have an account!
Thanks,
Nicoletta
Get Outlook for Android

 
From: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Sent: Thursday, December 12, 2019 10:35:39 AM
To: Nicoletta Fala <nfala@purdue.edu>
Cc: Pentchev, Valentin <vpentche@iu.edu>; Yan, Xiaoran <yan30@iu.edu>
Subject: RE: RE: Citation Network issues... 
 
Hi Nicoletta, 
 
Just to keep you updated, the compressed database file is still 50Gb so it’s too large to transfer using our usual file transfer system. I’m going to upload the file to AWS for you to download. Do you have an AWS account or should I make the file accessible to the world? 
 
Thanks, 
 
Matthew Hutchinson | INDIANA UNIVERSITY
Data Manager
IU Network Science Institute (IUNI)
1001 E SR 45/46 Bypass | Bloomington, IN  47408-1415
Email:  maahutch@iu.edu |Phone: (812) 855-1404
Fax: (812) 856-1192
 
From: Nicoletta Fala <nfala@purdue.edu> 
Sent: Wednesday, December 11, 2019 1:53 PM
To: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Cc: Pentchev, Valentin <vpentche@iu.edu>; Yan, Xiaoran <yan30@iu.edu>
Subject: [External] RE: RE: Citation Network issues...
 
This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.
 
Sounds great, thank you so much!
 
Nicoletta
 
Nicoletta Fala, Ph.D.
http://nicolettafala.com/  
 
 
From: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Sent: Wednesday, December 11, 2019 12:06:59 PM
To: Nicoletta Fala <nfala@purdue.edu>
Cc: Pentchev, Valentin <vpentche@iu.edu>; Yan, Xiaoran <yan30@iu.edu>
Subject: RE: RE: Citation Network issues... 
 
Hi Nicoletta, 
 
I have the database dump file ready to go and will send it to you tomorrow. Unfortunately, I work in a different office on campus on Wednesdays. Hopefully, I can send it to you first thing tomorrow. 
 
Thanks, 
 
Matthew Hutchinson | INDIANA UNIVERSITY
Data Manager
IU Network Science Institute (IUNI)
1001 E SR 45/46 Bypass | Bloomington, IN  47408-1415
Email:  maahutch@iu.edu |Phone: (812) 855-1404
Fax: (812) 856-1192
 
From: Nicoletta Fala <nfala@purdue.edu> 
Sent: Wednesday, December 11, 2019 11:26 AM
To: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Subject: [External] RE: RE: Citation Network issues...
 
This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.
 
Hi Matthew,
 
Do we have some sort of timeline for how long it takes to get things set up? We are trying to plan our research activities accordingly for the next few days so that we can complete our project before winter break. The graph database may be quicker, so I can work on that while the Postgres database form is going through. Dump file or VM for the graph db will work, whatever is easier/faster!
 
Thanks for helping us out!
 
Nicoletta
 
Nicoletta Fala, Ph.D.
http://nicolettafala.com/  
 
From: Nicoletta Fala
Sent: Tuesday, December 10, 2019 11:02 AM
To: Hutchinson, Matthew Alexander
Subject: RE: RE: Citation Network issues...
 
Hi Matthew,
 
I completed the form. It looks like we also want the graph database, if you can set us up with that 😊 
 
Thanks!
Nicoletta
 
Nicoletta Fala, Ph.D.
http://nicolettafala.com/  
 
From: Hutchinson, Matthew Alexander
Sent: Monday, December 9, 2019 4:39 PM
To: Nicoletta Fala
Cc: Witt, Michael C.
Subject: RE: RE: Citation Network issues...
 
If you are interested in using the Postgres database, you’ll need to complete the form here: http://iuni.iu.edu/resources/cadre/general-data-user and I’ll being the process of creating your IU computing account and granting you access to the database. If you’re interesting in using the Neo4j graph database, please let me know and I can send you a dump file or set yup a VM with the database. 
 
Thanks, 
 
Matthew Hutchinson | INDIANA UNIVERSITY
Data Manager
IU Network Science Institute (IUNI)
1001 E SR 45/46 Bypass | Bloomington, IN  47408-1415
Email:  maahutch@iu.edu |Phone: (812) 855-1404
Fax: (812) 856-1192
 
From: Nicoletta Fala <nfala@purdue.edu> 
Sent: Monday, December 9, 2019 4:35 PM
To: Hutchinson, Matthew Alexander <maahutch@iu.edu>
Cc: Witt, Michael C. <mwitt@purdue.edu>
Subject: [External] RE: RE: Citation Network issues...
 
This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.
 
Hi Matthew,
 
Thank you for explaining all of that. If I am understanding everything correctly, it is not possible to identify the references on a paper (i.e. the papers that the paper in question is citing) from CADRE outputs at this time. I therefore need to use the raw database to complete our project. I tried to follow the ISSI demos and I believe they all used the web interface—is there something special I need to do to get access to the postgres database you are referring to on the server so I can run my queries directly on that?
 
Best regards,
Nicoletta
 
 
From: Witt, Michael C.
Sent: Monday, December 9, 2019 3:57 PM
To: Nicoletta Fala
Subject: Fwd: RE: Citation Network issues...
 
Tldr; can you summarize please?
---------- Forwarded message ----------
From: "Yan, Xiaoran" <yan30@iu.edu>
Date: Dec 9, 2019 3:04 PM
Subject: RE: Citation Network issues...
To: "Hutchinson, Matthew Alexander" <maahutch@iu.edu>,"McGavin, Stephanie Andrea" <smcgavin@iu.edu>,"Witt, Michael C." <mwitt@purdue.edu>
Cc: "Pentchev, Valentin" <vpentche@iu.edu>,"Peli Kankanamalage, Chathuri Kamalasini Wimalasena" <cpelikan@iu.edu>
Hi Michael, 
 
Thank you for passing along the feedback. Matthew has explained everything quite clearly. We are aware of the confusing column names of the current output files. In fact, if you follow our ISSI demo, we manually rename these columns into “cited” and “citing” for better intuition.
 
If you do not mind making this communication public, I have created a GitHub issue for this matter.
https://github.com/iuni-cadre/Fellow1-data-citation/issues/2
 
Thanks!
 
Xiaoran
 
From: Hutchinson, Matthew Alexander <maahutch@iu.edu> 
Sent: Monday, December 9, 2019 2:13 PM
To: McGavin, Stephanie Andrea <smcgavin@iu.edu>; Yan, Xiaoran <yan30@iu.edu>; mwitt@purdue.edu
Cc: Pentchev, Valentin <vpentche@iu.edu>; Peli Kankanamalage, Chathuri Kamalasini Wimalasena <cpelikan@iu.edu>
Subject: RE: Citation Network issues...
 
This is an issue relating to how the database is structured and how the data is searched by the interface. I think it might be helpful if I try to clarify exactly what exactly is in the database st the current time. Please feel free to forward to Nicoletta if it’s helpful or I can work with her directly, if you would like me to. 
 
The graph database that is searchable via the web platform contains reference links but allows the user to search citations. 
 
If paper A has three items listed in its bibliography (b,c,d), in theory it should be connected to those eighty papers in the database following this structure: 
 
(a:paper)-[:REFERENCES]->(b|c|d:paper) 
 
This is because the data we receive from Microsoft is in a TSV structured like this: 
 
Paper Id 	Reference Paper Id
a	b
a	c
a	d
 
 
We load each row in this TSV into the graph database as an edge in the network. 
 
However, for the development version of the web interface we made the decision that for the type of users we anticipate for Cadre we expected that the ability to search citation networks would be a higher priority than the searching of reference networks. The result being that at the database level, the query effectively moves in reverse through the network and the data returned should reflect this.  
 
To use the cypher notation implemented by the database, a network query run by the user would look like this: 
 
(a|x|y:paper)-[:REFERENCES]->(b:paper{paper_title: “Paper the user is searching for” })
 
The tablular result the user receives in their jupyter environment should look like this: 
 
From	To
b	a
b	x
b	y
 
Even though the data is structured as a ‘reference’ network the user searches and receives the ‘citation’ network. 
 
In the case of Nicoletta’s query, I think she is expecting to receive the references of her initial paper but is actually receiving the citations. As to the missing citations, this is simply a result of Microsoft not having indexed those papers and included them in the data set. This is a persistent problem though this type of data. 
 
In general though, I wouldn’t recommend anyone using the web interface version of the data as a source for anything mission critical. It is still in development and has many unstable features and we can’t guarantee the accuracy of the results. I’d recommend using the postgres database on the server until we launch the stable version of the web interface next year. 
 
Hopefully this is helpful but it may be more detail than is required. We’re working on developing a more explicit network query building tool which will hopefully make this type of query much easier to construct. The current citation network tool is intended to be more of a ‘proof of concept’ than a useful feature of the interface. 
 
Let me know if I can add any more clarity. I’m also happy to share a copy of the graph database for direct use if that would be helpful. 
 
Matthew Hutchinson | INDIANA UNIVERSITY
Data Manager
IU Network Science Institute (IUNI)
1001 E SR 45/46 Bypass | Bloomington, IN  47408-1415
Email:  maahutch@iu.edu |Phone: (812) 855-1404
Fax: (812) 856-1192
 
From: McGavin, Stephanie Andrea <smcgavin@iu.edu> 
Sent: Monday, December 9, 2019 12:45 PM
To: Yan, Xiaoran <yan30@iu.edu>; Hutchinson, Matthew Alexander <maahutch@iu.edu>
Subject: FW: Citation Network issues...
 
Passing this on – will one of you reach out to Michael about this?
 
From: Witt, Michael C. [mailto:mwitt@purdue.edu] 
Sent: Monday, December 9, 2019 12:41 PM
To: McGavin, Stephanie Andrea <smcgavin@iu.edu>
Subject: [External] FW: Citation Network issues...
 
This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.
 
Hi Stephanie, we were feeling pretty good about our code and ran it and got some totally bogus results. We did some investigation and we’re finding bad data in the MAG Citation Network. See the two examples below. Can you please pass on to the team? Also please ignore my postdoc calling me high school principal. She starts a new job in January so she’s getting kind lippy with me. :-)
 
-
Michael Witt
Interim Associate Dean for Research
Associate Professor of Library Science
Purdue University Libraries and School of Information Studies
http://www.lib.purdue.edu/research/witt
 
From: Nicoletta Fala 
Sent: Monday, December 9, 2019 12:20 PM
To: Witt, Michael C. <mwitt@purdue.edu>
Subject: Citation Network issues...
 
I pulled a few threads, and here’s a problem that we’ll probably struggle with for a bit:
 
Background info: I have been using MAG with the Citation Network on CADRE. Selecting the citation network option gives you two extra spreadsheets—nodes and edges—which provide information on the references that the papers point to. Edges consists of two columns: From and To. For example, Paper A can point to Nodes (paper, dataset, …) 1, 2, 3, and Paper B can point to Nodes 3, 4, 5, 6. 
 
The problem: Not only does the citation network not provide all references, it also mixes up the From and To columns in edges. 
 
Example: I used one of my papers from 2014 as an example since I’m familiar with the references I should be finding. Here’s the paper: http://nicolettafala.com/files/Fala_et_al_EndAround_2015.pdf. It has 23 references. The three attached files are what CADRE gave me (no need to open them, attached in case you want them). 
I searched for my paper in the atqc.csv spreadsheet and found it under Paper ID 2537699788. I searched for Paper ID 2537699788 in the atcq_edges.csv spreadsheet and found one instance of it, under the “To” column. Paper ID 1515792823 is in the corresponding “From” column. Paper ID 1515792823 is a 2002 paper by Idris et al. that I referenced in my paper! So this particular row is flipped in the atcq_edges.csv spreadsheet. However, there are also 22 missing references from my paper. 
Paper ID 1515792823 happens to also be published in Air Traffic Control Quarterly, which means it appears in both atcq.csv and atcq_nodes.csv. So I tried to find the references Idris et al cited in this paper instead. This paper (https://pdfs.semanticscholar.org/cdcb/8f9380999147e162fbb4a67b9cfc889e8bc4.pdf) has 21 references. Searching for it in the atcq_edges.csv spreadsheet resulted in 108 instances, all in the “From” category. We already know that one is faulty because it’s my paper, so we’re down to 107, and we only expect 21, so somehow we have 80+ references that are not accounted for. Looking for some of these unaccounted nodes from the “To” column in the atcq_nodes.csv spreadsheet also result in papers as recent as 2018. 

Conclusion: This is where the problem starts, and I’m not sure there’s a valid way to correct for it given that some references are missing entirely and others are just wrong. Words of wisdom from High School Principal Witt? 
 
Side note: I also found a problem in my code—I was counting unique references. So, if the same journal had two papers that referenced the same dataset, I only counted it once. I fixed it over the weekend, but right now it’s a little irrelevant! 
 
Nicoletta
 
Nicoletta Fala, Ph.D.
http://nicolettafala.com/
